cpp/dataStructures/CMakeLists.txt

```diff
@@ -1,16 +1,12 @@
-cmake_minimum_required(VERSION 3.2.2)
-set(CMAKE_BUILD_TYPE Release)
-
-#For the shared library:
-set ( PROJECT_LINK_LIBS libcppCommon.dylib )
-link_directories( ../common/build )
+# However, the file(GLOB...) allows for wildcard additions:
+file(GLOB SOURCES "src/*.cpp")
 
-#Bring the headers, such as Student.h into the project
-include_directories(include)
-include_directories(../common/include/)
+# Generate the shared library from the sources
+add_library(cppDS SHARED ${SOURCES})
 
```

Comment:
Got it. If I can bother you with another question: When I download the project on windows and run it using mingw-make.exe, it builds the dependent library and testDS.exe. However, when I run testDS.exe, it's unable to find the dependency DLL. I see the dlls being generated in build/common and build/dataStructures but it seems like the executable cannot find these. What's a good way to fix this ? ( I don't want it to break for linux/macosx)